app.rb

```diff
@@ -0,0 +1,177 @@
+require './person'
+require './book'
+require './student'
+require './classroom'
+require './rental'
+require './teacher'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    puts "Welcome to School library App!\n\n"
+  end
+
+  def run
+    loop do
+      puts '=' * 60
+      puts display_options
+      puts '=' * 60
+      print 'Your option: '
+      option = user_option
+      puts option == 'Invalid option' ? 'Invalid option, please type correct number!' : available_options(option)
+      # puts try
+    end
+  end
+
+  def display_options
+    'Please choose an option by entering a number between 1 to 7:
+    1 - List of all books
+    2 - List of all people
+    3 - Create a person
+    4 - Create a book
+    5 - Create a rental
+    6 - List of all rental for given id
+    7 - Exit'
+  end
+
+  def user_option
+    user_choice = gets.chomp.to_i
+    user_choice.positive? && user_choice <= 7 ? user_choice : 'Invalid option'
+  end
+
+  def available_options(user_choice)
+    case user_choice
+    when 1
+      list_books(@books)
+    when 2
+      list_people(@people)
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      create_rental(@people, @books)
+    when 6
+      list_rentals(@rentals)
+    else
+      puts 'Thank you for using this app!'
+      exit
+    end
+  end
```

Comment:
- [ ] As stated in the project requirement, you are expected to have the above-highlighted code in the `main.rb` file and not `app.rb` file. 

See project requirements below:
![image](https://user-images.githubusercontent.com/84156714/216698997-6d04f1b5-9cbf-45fe-b196-3779547e7efe.png)

- Kindly refactor your code to meet this requirement. You may not follow my code exactly particularly the fact that I like the way you broke down your code. Essentially, ensure that the `main.rb` file does all this things stated in the requirement. See the example below: 

```ruby
# main.rb
def main
  app = App.new()

  response = nil

  # *Present the user with a list of options to perform.
  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    # *Lets users choose an option.
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      # *Have a way to quit the app.
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

# 'main' method should be invoked at the end of the main.rb file, hint in this lesson https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point
main()
```